code/__HELPERS/stoplag.dm

```diff
@@ -5,17 +5,36 @@
 
 /// Returns the number of ticks slept
 /proc/stoplag(initial_delay)
+	//No master controller active, sleep for the tick lag to allow other things to run
 	if (!Master || !(Master.current_runlevel & RUNLEVELS_DEFAULT))
 		sleep(world.tick_lag)
 		return 1
+	//Set the default initial delay, if one isn't provided
 	if (!initial_delay)
 		initial_delay = world.tick_lag
 	. = 0
+	//Begin tracking if we are in debugging
+#ifdef TICK_OVERRUN_DEBUG
+	var/stack_traced = FALSE
+	var/should_print = !GLOB.controller_frozen
+#endif
```

Comment:
Not sure what you mean, both variables are used. stack_traced is set locally inside this proc and should_print needs to be recorded for sleeping because if it updated, then we would get things that ran after this triggered.